h/features/models.py

```diff
@@ -105,3 +112,57 @@ def remove_old_flags(cls):
 
     def __repr__(self):
         return '<Feature {f.name} everyone={f.everyone}>'.format(f=self)
+
+
+class Cohort(Base, mixins.Timestamps):
+    __tablename__ = 'cohort'
+
+    id = sa.Column(sa.Integer, autoincrement=True, primary_key=True)
+    name = sa.Column(sa.UnicodeText(), nullable=False, index=True)
+
+    # Cohort membership
+    members = sa.orm.relationship(
+        'User', secondary='user_cohort', backref=sa.orm.backref(
+            'cohorts', order_by='Cohort.name'))
+
+    def __init__(self, name, creator):
+        self.name = name
+        self.creator = creator
+        self.members.append(creator)
+
+    @sa.orm.validates('name')
+    def validate_name(self, key, name):
+        if not COHORT_NAME_MIN_LENGTH <= len(name) <= COHORT_NAME_MAX_LENGTH:
+            raise ValueError('name must be between {min} and {max} characters '
+                             'long'.format(min=COHORT_NAME_MIN_LENGTH,
+                                           max=COHORT_NAME_MAX_LENGTH))
+        return name
+
+    def __repr__(self):
+        return '<Cohort: %s>' % self.slug
+
+    @property
+    def slug(self):
+        """A version of this group's name suitable for use in a URL."""
+        return slugify.slugify(self.name)
+
+    @classmethod
+    def get_by_id(cls, id_):
+        """Return the cohort with the given id, or None."""
+        try:
+            return cls.query.filter(cls.id == id_).one()
+        except exc.NoResultFound:
+            return None
+
+
+USER_COHORT_TABLE = sa.Table(
```

Comment:
So in addition to the suggestion for naming this the other way around (see above), we need to not make the same mistake we made with the `user_group` table, namely:

- we should have a uniqueness constraint on `(cohort_id, user_id)`
- the table should have a PK column

The former because >1 rows for any user+cohort pair makes removal harder than it should be. The latter because fixing problems like that becomes harder when we don't have a PK to sort by.